### PR TITLE
little review #61

### PR DESCRIPTION
**Pull request by QuentinPerez**
- use snprintf instead of sprintf
- use recv instead of read
- use send instead of write
- close sockfd when the program fails
- add debug with strerror(errno)

### CODE REVIEW

File: scw-boot-tools/scw-update-server-state.c [hunk 1]
```diff
@@ -38,44 +44,40 @@ Content-Length: %d\r\n\r\n\
  retry:
   while (retries-- > 0) {
     //printf("Retries: %d\n", retries);
+    errno = 0;
     sockfd = socket(AF_INET, SOCK_STREAM, 0);
     if (sockfd < 0) {
-      fprintf(stderr, "ERROR opening socket... ");
-      goto retry;
+      RETRY("opening socket...");
```
Comment (moul) [inline scw-boot-tools/scw-update-server-state.c]: `close(sockfd)` with `sockfd < 0` ?

Comment (QuentinPerez) [inline scw-boot-tools/scw-update-server-state.c]: Oops Copy-Paste I fix it

**Comment by QuentinPerez** [comment]
Fixed :+1:

**Comment by moul** [comment]
LGTM

**Pull request merged**